compiler/optimizations/loopInvariantCodeMotion.cpp

```diff
@@ -1261,6 +1261,9 @@ void loopInvariantCodeMotion(void) {
   // compute array element alias sets
   computeNoAliasSets();
 
+  // optimize certain statements in foralls to unordered
+  optimizeForallUnorderedOps();
```

Comment:
Not sure how I missed this before, but why are we running this as part of LICM?

Reply:
I don't generally add passes to the compiler (because it slows everything down, and requires test updates). It needs to run late in compilation. I could move it before/after LICM proper but this might better be future work at this point.